server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -834,18 +834,15 @@ export class AgenticChatController implements ChatHandlers {
                 if (!availableToolNames.includes(toolUse.name)) {
                     throw new Error(`Tool ${toolUse.name} is not available in the current mode`)
                 }
+
+                // remove progress UI
+                await chatResultStream.removeResultBlockAndUpdateUI(progressPrefix + toolUse.toolUseId)
```

Comment:
wouldn't this break other tool?

Reply:
This only removes the messages that has the progressPrefix, which is the in progress rendered UX. When the streaming result is finalized, we no longer need these in progress ux.